### PR TITLE
[ASTGen] Cleanup type attribute generation

### PR DESCRIPTION
**Pull request by rintaro**
Use similar scheme as DeclAttribute.
* Create `BridgedTypeAttribute.createSimple()` and `BridgedTypeAttributes.add()`, instead of `BridgedTypeAttributes.addSimple()`
* Create `DeclAttributes::createSimple()` to align with `TypeAttribute`

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: include/swift/AST/ASTBridging.h [hunk 1]
```diff
@@ -1439,78 +1439,33 @@ void BridgedStmt_dump(BridgedStmt statement);
...
+  BridgedTypeAttrKind_None,
 };
 
 SWIFT_NAME("BridgedTypeAttrKind.init(from:)")
 BridgedTypeAttrKind BridgedTypeAttrKind_fromString(BridgedStringRef cStr);
 
-SWIFT_NAME("BridgedTypeAttributes.init(context:)")
-BridgedTypeAttributes BridgedTypeAttributes_create(BridgedASTContext cContext);
+SWIFT_NAME("BridgedTypeAttributes.init()")
+BridgedTypeAttributes BridgedTypeAttributes_create();
```
Comment (hamishknight) [inline include/swift/AST/ASTBridging.h:1452]: I know this was the case before, but should we expose this to Swift as a static method named `new` or something to make it clear it's doing a manual allocation?

File: lib/AST/ASTBridging.cpp [hunk 2]
```diff
@@ -1945,53 +1941,65 @@ BridgedYieldStmt BridgedYieldStmt_createParsed(BridgedASTContext cContext,
...
-BridgedTypeAttributes BridgedTypeAttributes_create(BridgedASTContext cContext) {
-  return {new TypeAttributes(cContext.unbridged())};
+static llvm::Optional<TypeAttrKind> unbridged(BridgedTypeAttrKind kind) {
+  switch (kind) {
+#define TYPE_ATTR(SPELLING, _)                                                 \
+  case BridgedTypeAttrKind_##SPELLING:                                         \
+    return TAK_##SPELLING;
+#include "swift/AST/Attr.def"
+  case BridgedTypeAttrKind_None:
+    return {};
```
Comment (hamishknight) [inline lib/AST/ASTBridging.cpp:1963]:
```suggestion
    return llvm::None;
```

**Review by hamishknight — APPROVED** [review]

**Review by ahoppen — COMMENTED** [review]
Swift changes look good to me

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Pull request merged**